adoc/chapters/programming_interface.adoc

```diff
@@ -5186,718 +5173,637 @@ If no template argument is provided, then the default allocator for the SYCL
+[source,role=synopsis,id=api:buffer-get-allocator]
 ----
-template <access_mode Mode = access_mode::read_write,
-          target Targ = target::device>
-accessor<T, Dimensions, Mode, Targ> get_access(handler& commandGroupHandler,
-                                               range<Dimensions> accessRange,
-                                               id<Dimensions> accessOffset = {})
+AllocatorT get_allocator() const;
 ----
-   a@ Returns a valid [code]#accessor# to the buffer with the specified access
-      mode and target in the command group buffer.  The accessor is a
-      <<ranged-accessor>>, where the range starts at the given offset from the
-      beginning of the buffer.  The value of target can be
-      [code]#target::device#, [code]#target::constant_buffer# or
-      [code]#target::host_task#.
+_Returns:_ The allocator provided to the buffer.
 
-Throws an [code]#exception# with the [code]#errc::invalid# error code if
-the sum of [code]#accessRange# and [code]#accessOffset# exceeds the range of
-the buffer in any dimension.
+'''
 
-a@
-[source]
+.[apititle]#buffer::get_access#
+[source,role=synopsis,id=api:buffer-get-access]
 ----
-template <access_mode Mode>
-accessor<T, Dimensions, Mode, target::host_buffer>
-get_access(range<Dimensions> accessRange, id<Dimensions> accessOffset = {})
-----
-   a@ Deprecated in SYCL 2020.  Use [code]#get_host_access()# instead.
+template <access_mode Mode = access_mode::read_write,                          (1)
+          target Targ = target::device>
+accessor<T, Dimensions, Mode, Targ> get_access(handler& commandGroupHandler);
 
-Returns a valid host [code]#accessor# to the buffer with the specified access
-mode and target.  The accessor is a <<ranged-accessor>>, where the range starts
-at the given offset from the beginning of the buffer.  The value of target can
-only be [code]#target::host_buffer#.
+template <access_mode Mode = access_mode::read_write,                          (2)
+          target Targ = target::device>
+accessor<T, Dimensions, Mode, Targ>
+get_access(handler& commandGroupHandler, range<Dimensions> accessRange,
+           id<Dimensions> accessOffset = {});
 
-Throws an [code]#exception# with the [code]#errc::invalid# error code if
-the sum of [code]#accessRange# and [code]#accessOffset# exceeds the range of
-the buffer in any dimension.
+template <typename... Ts> auto get_access(Ts...);                              (3)
 
-a@
-[source]
+template <typename... Ts> auto get_host_access(Ts...);                         (4)
 ----
-template <typename... Ts> auto get_access(Ts... args)
-----
-   a@ Returns a valid [code]#accessor# as if constructed via passing the buffer
-      and all provided arguments to the [code]#accessor# constructor.
+_Returns (1):_ A valid [code]#accessor# to the buffer with the specified access
+mode and target in the command group buffer.
+The value of target can be [code]#target::device#,
+[code]#target::constant_buffer# or [code]#target::host_task#.
```

Comment:
Should these be preconditions or constraints?

Reply:
Great, thanks! That has been addressed.